src/dmd/todt.d

```diff
@@ -543,9 +543,9 @@ extern (C++) void Expression_toDt(Expression e, DtBuilder dtb)
                 e.fd.vthis = null;
             }
             Symbol *s = toSymbol(e.fd);
-            if (e.fd.isNested())
+            if (e.fd.needsClosure())
             {
-                e.error("non-constant nested delegate literal expression `%s`", e.toChars());
+                e.error("delegate literal expression `%s` requires closure", e.toChars());
```

Comment:
Not covered.

Reply:
True, I dont know what would trigger it though. I'll just remove it.